superduper/components/streamlit.py

```diff
@@ -0,0 +1,45 @@
+import typing as t
+
+from superduper import Component
+
+
+class Streamlit(Component):
+    """Streamlit demo function to be deployed on the streamlit server.
+    
+    :param demo_func: Callable which builds the demo.
+    :param default: Set to `True` if this is to be the frontpage.
+    """
+    type_id: t.ClassVar[str] = 'streamlit'
+    demo_func: t.Callable
+    default: bool = False
+
+    @property
+    def page(self):
+        """Get the streamlit page for the multi-page app."""
+        import streamlit as st
+        return st.Page(self.demo_func, title=self.identifier, default=self.default)
```

Comment:
The demo_func should take `db, *args, **kwargs`, for example

```python
def demo_func(db **kwargs):
    model = db.load("xxx")
    # ....
```

```python
def page(self):
    def page_func():
        self.demo_func(self.db)
    return st.Page(page_func, title=..., default=...)
```



WDYT?